plugins/modules/credential.ps1

```diff
@@ -0,0 +1,126 @@
+$provider_name = $module.Params.provider_name
+$state = $module.Params.state
+$force = $module.Params.force
+$checkMode = $module.CheckMode
+
+$module.Result.changed = $false
+
+try {
+    $getCredendtialSplat = @{
+        SqlInstance = $sqlInstance
+        SqlCredential = $sqlCredential
+        Identity = $identity
+        EnableException = $true
+    }
+    $existingCredential = Get-DbaCredential @getCredendtialSplat
+
+    if ($state -eq "absent") {
+        # Remove credential if it exists
+        if ($null -ne $existingCredential) {
+            try {
+                $removeCredentialSplat = @{
+                    #SqlInstance = $sqlInstance
+                    #SqlCredential = $sqlCredential
+                    #Identity = $identity
+                    EnableException = $true
+                    WhatIf = $checkMode
+                    Confirm = $false
+                }
+                $output = $existingCredential | Remove-DbaCredential @removeCredentialSplat
+                $module.Result.changed = $true
+            }
+            catch {
+                $module.FailJson("Removing credential failed: $($_.Exception.Message)", $_)
+            }
+        }
+    }
+    elseif ($state -eq "present") {
+        # Create credential
+        if (($null -ne $existingCredential -and $force -eq $true) -or $null -eq $existingCredential) {
+            try {
+                $newCredentialSplat = @{
+                    SqlInstance = $sqlInstance
+                    SqlCredential = $sqlCredential
+                    Identity = $identity
+                    EnableException = $true
+                    WhatIf = $checkMode
+                    Force = $force
+                    Confirm = $false
+                }
+                if ($null -ne $name) {
+                    $newCredentialSplat.Add("Name", $name)
+                }
+                if ($null -ne $secure_password) {
+                    $newCredentialSplat.Add("SecurePassword", $secure_password)
+                }
+                if ($null -ne $mapped_class_type) {
+                    $newCredentialSplat.Add("MappedClassType", $mapped_class_type)
+                }
+                if ($null -ne $provider_name) {
+                    $newCredentialSplat.Add("ProviderName", $provider_name)
+                }
+                $output = New-DbaCredential @newCredentialSplat
+                $module.result.changed = $true
+            }
+            catch {
+                $module.FailJson("Creating credential failed: $($_.Exception.Message)", $_)
+            }
+        }
+        # Return existing credential if nothing is changed
+        else {
+            try {
+                $output = Get-DbaCredential @getCredendtialSplat
+                $module.result.changed = $false
+            }
+            catch {
+                $module.FailJson("Configuring credential failed: $($_.Exception.Message)", $_)
+            }
```

Comment:
```suggestion
        elseif ($null -ne $existingCredential) {
            $output = $existingCredential
```
Might be cleaner & bump code coverage to just re-use the $existingCredential here